SetupDataPkg/Include/Library/PlatformConfigDataLib.h

```diff
@@ -23,6 +23,6 @@ extern PROFILE    gProfileData[];
 // this does not count the generic profile, which is not
 // in gProfileData, but rather in gKnobData's defaults
 extern UINTN  gNumProfiles;
-extern CHAR8  *gProfileFlavorName;
+extern CHAR8  *gProfileFlavorName[];

```

Comment:
If this is a pointer to separate strings, I recommend the name be `gProfileFlavorNames`. I think that would have made this a little more obvious the first time.

Reply:
A minor detail, but looking back at the previous PR (#207), specifically [SetupDataPkg/Library/PlatformConfigDataLibNull/PlatformConfigDataLibNull.c](https://github.com/microsoft/mu_feature_config/blob/main/SetupDataPkg/Library/PlatformConfigDataLibNull/PlatformConfigDataLibNull.c), we usually don't use `0` for `NULL` per the EDK II style guide.

For struct initializers (like `gKnobData`) that contain a mix of primitive vs derived member types zero is fine. Since `gProfileFlavorName` is solely an array of pointers, it would align more closely with our normal pointer `NULL` usage to use:

```c
CHAR8  *gProfileFlavorName = { NULL };
```

Reply:
Agreed. I actually ended up changing the c file to make it the same as header file definition. Will do another round of sanity check before merging this.

Reply:
Verified on pipeline build and platform builds.